File: app/src/main/java/com/divya/rememberyourpasswords/UserViewModel.java
package com.divya.rememberyourpasswords;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.content.Context;
import android.support.annotation.NonNull;

import com.divya.rememberyourpasswords.data.UserAccountDatabase;
import com.divya.rememberyourpasswords.data.UserRepository;

/**
 * Created by divya on 5/16/2018.
 */

public class UserViewModel extends ViewModel {

    private UserRepository userRepository;

    public UserViewModel(Context context) {

        userRepository = UserRepository.getInstance(UserAccountDatabase.getAppDatabase(context).userAccountDao());
    }

    void createUser(String username, String password)
    {
        userRepository.insertUser(username, password);
    }

    boolean checkValidLogin(String username, String password)
    {
        return userRepository.isValidAccount(username, password);
    }

   public static class Factory implements ViewModelProvider.Factory {
        private final Context ctxt;

        Factory(Context ctxt) {
            this.ctxt=ctxt.getApplicationContext();
        }

        @NonNull
        @Override
        public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
            return((T)new UserViewModel(ctxt));
        }
    }
}

File: Kotlin/app/src/main/java/com/divya/signup_login_kotlin/UserViewModel.kt
package com.divya.signup_login_kotlin

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import android.content.Context

import com.divya.rememberyourpasswords.data.UserAccountDatabase
import com.divya.rememberyourpasswords.data.UserRepository

/**
 * Created by divya on 5/16/2018.
 */

class UserViewModel(context: Context) : ViewModel() {

    private val userRepository: UserRepository

    init {

        userRepository = UserRepository.getInstance(UserAccountDatabase.getAppDatabase(context).userAccountDao())
    }

    internal fun createUser(username: String, password: String) {
        userRepository.insertUser(username, password)
    }

    internal fun checkValidLogin(username: String, password: String): Boolean {
        return userRepository.isValidAccount(username, password)
    }

    class Factory internal constructor(ctxt: Context) : ViewModelProvider.Factory {
        private val ctxt: Context

        init {
            this.ctxt = ctxt.applicationContext
        }

        override fun <T : ViewModel> create(modelClass: Class<T>): T {
            return UserViewModel(ctxt) as T
        }
    }
}
